### ISSUE TITLE
Mobile Safari on iOS not getting cleaned up after app or tab terminates

### ISSUE THREAD

**Issue by Flamenco** [OpenVidu/openvidu#691]
I have been having issues with mobile safari not disconnecting when tab or app is closed for well over a year.  

I will start out by explaining what causes the issue.

When the app (Safari) is quit from the list of iOS applications, or the tab is closed from Safari's tab overview, my other clients never receive a session or connection terminated message.  In fact, they do not even get the ping timeout message.  I have confirmed that a message is never received via the connection websocket.

For starters, this is not happening using iOS Chrome.  Only Safari.  And all version up to the current one.  It has been happening when using the most current OpenVidu browser and OpenVidu server, and going back to at least 4 versions.

In my client, i have hooked into every possible disconnect message, including `beforeunload`, `visibilitychange`, and `pagehide`, etc;  I have even tried sending messages to disconnect to my backend using sendBeacon and fetch with keepalive.  But Safari just ditches every API callback when closing.

So in the end, because my application creates a websocket to my server, I monitor when that websocket closes, and send a message to my media server with a browser id that I include in my openvidu connections.  The media server backend checks every connection for that id and closes them.  This results in my clients closing the video players within milliseconds of the iOS browser closing.  So even better than waiting for a ping timeout.

So 2 issues here.

1.  Any idea why I am not getting ping timeout from iOS mobile Safari closing, but getting one from iOS Chrome closing?

2. Why does the OpenVidu websocket for the connection not just detect a close and send a disconnect message like I am doing?  It seems like that would be a helpful option when creating the connection at the very least, and result in a much more responsive experience for users.

**Comment by CSantosM** [comment]
Hi @Flamenco , 

I've tested the same test cases for trying to reproduce your issue. I've connected to the session with two participant (Chrome browser and iPhone 7 v15.4 Safari ) and force quit the Safari Browser app.
 Everything is working fine, the Chrome participant is notifyed that the other participant is not in the session after 10 seconds approximately. Same behaviour closing the webapp tab. 

I can ensure that this is working properly. I would recommend you test with our https://demos.openvidu.io  deployment for testing your issue and let us know
